### ISSUE TITLE
Improve action feedback for nudging items/envelope points, and growing/shrinking item edges

### ISSUE THREAD

**Issue by ScottChesworth** [jcsteh/osara#155]
Currently, when nudging items/envelope points and growing/shrinking item edges, action feedback reports a summary of the action title - "shrink left edges of items" or "move items/envelope points right", for example. However, this seems like it duplicates functionality already made available via shortcut help, and it doesn't report positional information.

Can this feedback be tweaked to report changes to positional information as the item edge grows/shrinks or the envelope point moves? I'm imagining hearing spoken feedback in the same format as scrubbing.

If so, would it also be possible to either A) add check boxes to OSARA prefs to toggle action feedback for each of the relevant categories, or B) reword the existing check box that toggles feedback when scrubbing to be something like "Report position on timeline when scrubbing, nudging, growing and shrinking"? The jury is still out on whether a combined check box would be popular among users, but figured I'd include the idea here because I know you're keen to avoid clutter (and also, seeing as the existing check box is the default position when opening the prefs dialog, I'm sure it'd be something the troops could learn to live with).

One thing I haven't figured out is how positional information would be reported productively if multiple items or points were selected. It's a brain-breaker. Any thoughts/suggestions much appreciated.

FYI, the proposed change would apply to the following actions:
Item edit: Grow left edge of items (40225)
Item edit: Grow right edge of items (40228)
Item edit: Shrink left edge of items (40226)
Item edit: Shrink right edge of items (40227)
Item edit: Move items/envelope points left (40120)
Item edit: Move items/envelope points right (40119)
Item edit: Move items/envelope points left by grid size (40793)
Item edit: Move items/envelope points right by grid size (40794)
Item edit: Move items left, preserving timing of contents (40121)
Item edit: Move items right, preserving timing of contents (40122)
Time selection: Nudge left (40039)
Time selection: Nudge right (40040)
Time selection: Nudge left edge left (40320)
Time selection: Nudge left edge right (40321)
Time selection: Nudge right edge left (40322)
Time selection: Nudge right edge right (40323)

I realise that's a boat load of actions and that it might not be feasible with the time you currently have, but thanks in advance for anything you can do here.

**Comment by jcsteh** [comment]
In order to really consider this, each possible scenario needs to be
considered and appropriate output determined (preferably with examples for
clarity). Is it really useful to report, for example, "Item points moved 2%
left"? The user presumably configured that value, so I don't follow why
this is useful. In contrast, scrubbing reports the new position on the
timeline; the user couldn't know the position without checking it. On the
flip side, is reporting the new position (or edge) useful? That would
result in a ridiculous amount of verbosity, since you can't just report a
relative value; saying "left edge moved to 2%" might suggest it moved right
to the start of the project. And then there's the question of having
multiple items or points selected, and that just starts getting insane.
This is basically why I didn't include this kind of feedback in the first
place, along with the fact that calculating all of this is tricky and might
impact performance.

Long story short: if people really want this, someone is going to need to
clearly specify the user experience for every possible scenario. From
there, we can work out whether it's technically feasible.

**Comment by ScottChesworth** [comment]
Cool, I'll keep collecting feedback and thinking on it.

**Comment by RDMurray** [comment]
see #225 and #227 for an implementation of some of these actions.  It reports the position based on the last item spoken by Osara.  It currently doesn't have a verbosity option, but one could be added if it is considered necessary.

**Comment by ScottChesworth** [comment]
@RDMurray's implementations from #225 and #227 have been out in the wild for a fair old while now with no reports of related performance issues that I'm aware of. My hope is that now we're sure his approach doesn't impact upon performance, perhaps we can carry it over to "Item edit: Move items/envelope points left (ID 40120)" and "Item edit: Move items/envelope points right (ID 40119)", the only two actions listed in the OP that are still giving generic feedback. Can we report the new positions of left and right item edges and tweak from there if necessary, perhaps? I know it's extra verbiage, but there are SWS variants of those actions without feedback implemented that power users can switch to if they'd prefer to nudge in silence, and my thinking is that it'd still be progress. Context relevant verbiage is still preferable over generic verbiage innit?

**Comment by RDMurray** [comment]
Is it necessary to report both edge positions?  I think I would have trouble focusing on the one I was interested in while ignoring the other.  Would the same feedback as "Time selection: Nudge right" etc be okay?

**Comment by RDMurray** [comment]
The current feedback for grow/shrink edge of items is based on Osara's idea of the focused item, even if multiple items are selected.  Would that be okay here?  I suppose it could also say the number of items effected.  Either way these commands should be consistent and would probably share the same code.
